### ISSUE TITLE
Support unified cgroups (cgroups v2)

### ISSUE THREAD

**Issue by dmazhar-cogniance** [aws/containers-roadmap#1535]
### Summary
It looks like cgroup v2 is not supported by ecs-agent. Getting this error in logs:
```
cloudwatch metrics for container XXX not collected, reason (cpu): need at least 2 data points in queue to calculate CW stats set" module=engine.go
```
### Description
Hi team! We're using ecs-agent with flatcar OS, and recent beta migrated to the cgroups v2 which broke the cloudwatch metrics for our ECS clusters. Could you please suggest if cgroup v2 is supported by ecs-agent, or are there any plans to support it? For now we're locked out OS version, but need to plan what to do next.

**Comment by Phylu** [comment]
I have the issue as follows:

- The host system is flatcar stable which I install as an AMI from the AWS marketplace.
- I upgraded flatcar to a version newer than 2983.2.0 [1] which enables Groups V2 by default after it has been added to the Linux kernel in 2014 and is adopted by Linux distributions since 2019 [2].
- The ECS agent does not support CGroups 2 and therefore the metrics collection fails [3].
- I see memory leaks where the ECS agent uses multiple GB of memory on the host system so that the system degrades and I loose service connection sporadically. This is even worse than the issue with the missing metrics.
 
The problem with locking the flatcar to a working version that predates October 2021 is, that it contains a multitude of security vulnerabilities that have ben resolved in the latest stable version.

I really hope, that this can get some priority accordingly.
 
[1] https://www.flatcar-linux.org/releases/#release-2983.2.0
[2] https://medium.com/nttlabs/cgroup-v2-596d035be4d7
[3] https://github.com/flatcar-linux/Flatcar/issues/585
